src/modulus_convert.rs

```diff
@@ -0,0 +1,548 @@
+            .collect()
+    }
+
+    pub fn mult_r1_r2_helper3<R: RngCore>(
+        mut rng_3_1: R,
+        p: u64,
+        batch_size: usize,
+    ) -> Vec<ReplicatedSecretSharing> {
+        let mut output_shares = Vec::with_capacity(batch_size);
+        for _i in 0..(batch_size - 1) {
+            let s_3_1 = rng_3_1.gen_range(0..p);
+            output_shares.push(ReplicatedSecretSharing { sh1: 0, sh2: s_3_1 });
+        }
+        output_shares
+    }
+
+    pub fn xor_r1_r2_r3_helper1<R: RngCore>(
+        mut rng_1_2: R,
+        r_binary_shares: &[ReplicatedBinarySecretSharing],
+        shares_of_r1r2: &[ReplicatedSecretSharing],
+        received_values: &[u64],
+        p: u64,
+        batch_size: usize,
+    ) -> Vec<ReplicatedSecretSharing> {
+        let mut output_shares = Vec::with_capacity(batch_size);
+        for i in 0..(batch_size - 1) {
+            let alpha_1 = sub_mod_p(
+                u64::from(r_binary_shares[i].sh1),
+                mult_mod_p(2, shares_of_r1r2[i].sh1, p),
+                p,
+            );
+            let alpha_2 = sub_mod_p(
+                u64::from(r_binary_shares[i].sh2),
+                mult_mod_p(2, shares_of_r1r2[i].sh2, p),
+                p,
+            );
+            let s_1_2: u64 = rng_1_2.gen_range(0..p);
+
+            output_shares.push(ReplicatedSecretSharing {
+                sh1: sub_mod_p(alpha_1, mult_mod_p(2, received_values[i], p), p),
+                sh2: sub_mod_p(alpha_2, mult_mod_p(2, s_1_2, p), p),
+            });
+        }
+        output_shares
+    }
+
+    pub fn xor_r1_r2_r3_helper2<R: RngCore>(
+        mut rng_1_2: R,
+        mut rng_2_3: R,
+        r_binary_shares: &[ReplicatedBinarySecretSharing],
+        shares_of_r1r2: &[ReplicatedSecretSharing],
+        p: u64,
+        batch_size: usize,
+    ) -> (Vec<ReplicatedSecretSharing>, Vec<u64>) {
+        // To send to helper 3
+        let mut values_to_share = Vec::with_capacity(batch_size);
+        let mut output_shares = Vec::with_capacity(batch_size);
+        for i in 0..(batch_size - 1) {
+            let alpha_2: u64 = sub_mod_p(
+                u64::from(r_binary_shares[i].sh1),
+                mult_mod_p(2, shares_of_r1r2[i].sh1, p),
+                p,
+            );
+            let s_1_2: u64 = rng_1_2.gen_range(0..p);
+            let s_2_3: u64 = rng_2_3.gen_range(0..p);
+            let d_2: u64 = sub_mod_p(alpha_2 * u64::from(r_binary_shares[i].sh2), s_1_2, p);
+            output_shares.push(ReplicatedSecretSharing {
+                sh1: sub_mod_p(alpha_2, mult_mod_p(2, s_1_2, p), p),
+                sh2: sub_mod_p(
+                    u64::from(r_binary_shares[i].sh2),
+                    mult_mod_p(2, add_mod_p(d_2, s_2_3, p), p),
+                    p,
+                ),
+            });
+            values_to_share.push(d_2);
+        }
+        (output_shares, values_to_share)
+    }
+
+    pub fn xor_r1_r2_r3_helper3<R: RngCore>(
```

Comment:
```suggestion
    pub fn xor_r1_r2_r3_helper3<R: Rng>(
```

You are using Rng functions.